lib/hanami/utils/inflector.rb

```diff
@@ -27,17 +27,31 @@ def add(key, value)
         # @since 0.6.0
         # @api private
         def ===(other)
-          key = other.downcase
+          key = if other.downcase =~ /_(\w*)\z/
```

Comment:
You may want to change `\w` (A word character `[a-zA-Z0-9_]`) with `[[:alpha:]]` (Alphabetic character), to match chars beyond ASCII set.